_stbt/diff.py

```diff
@@ -49,20 +49,77 @@ def __repr__(self) -> str:
                 self.motion, self.region, _frame_repr(self.frame)))
 
 
-class FrameDiffer():
-    """Interface for different algorithms for diffing frames in a sequence.
+class UNSET:
+    """Sentinel"""
 
-    Say you have a sequence of frames A, B, C. Typically you will compare frame
-    A against B, and then frame B against C. This is a class (not a function)
-    so that you can remember work you've done on frame B, so that you don't
-    repeat that work when you need to compare against frame C.
+
+class Differ:
     """
-    def diff(self, frame: FrameT):
+    This is a low level interface enabling customising how two images are
+    compared.  It's used by `wait_for_motion` and `press_and_wait`.
+
+    `Differ` is designed so we can efficiently implement `GrayscaleDiff`
+    without doing the colour -> grayscale conversion on the same image twice -
+    just as we could before.  To avoid internal state the interface makes
+    preprocessing explicit.  This makes the classes hard to use, but that's ok -
+    it's a low-level interface intended mostly for internal use.
+
+    The only public interface is the subclass constructors.  The methods are not
+    intended for use by external code.
```

Comment:
Are we going to support users implementing their own Differ subclasses? If so, they'll have to implement `preprocess` and `diff`, so those are technically public interfaces too.

Reply:
Probably not — the interface is too complex. The real reason for making the base class public is for type annotations.